CacheMeIfYouCan/CacheMeIfYouCan/CacheTypes/FileSystemDataCache.swift

```diff
@@ -0,0 +1,128 @@
+//
+//  FilesystemDataCache.swift
+//  CacheMeIfYouCan
+//
+//  Created by Ellen Shapiro on 12/10/18.
+//  Copyright © 2018 Bakken & Baeck. All rights reserved.
+//
+
+import Foundation
+
+open class FileSystemDataCache<T: DataConvertible>: Cache {
+    
+    public typealias StoredType = T
+    
+    let folderURL: URL
+    
+    public lazy var localQueue = FileSystemDataCache.defaultQueue
+    
+    public init(folderURL: URL) {
+        self.folderURL = folderURL
+        
+        do {
+            try FileManagerHelper.createFolderIfNeeded(at: folderURL)
+        } catch {
+            LogHelper.log("Error creating folder: \(error)")
+        }
+    }
+    
+    private func localURL(for url: URL) -> URL {
+        let fileName = FilesystemPathHelper.fileName(from: url)
+        return self.folderURL
+            .appendingPathComponent(fileName)
+    }
+    
+    open func actuallyStore(item: T, for url: URL) throws {
+        guard let data = item.toData else {
+            return
+        }
+        
+        let url = self.localURL(for: url)
+        try data.write(to: url)
+    }
+    
+    open func actuallyFetchItem(for url: URL) throws -> T? {
+        let url = self.localURL(for: url)
+        guard FileManagerHelper.fileExists(at: url) else {
+            return nil
+        }
+    
+        let data = try Data(contentsOf: url)
+        let item = T(data: data)
+
+        return item
+    }
+    
+    open func clearAll() throws {
+        try FileManagerHelper.removeContentsOfFolder(at: self.folderURL)
```

Comment:
It looks like if you `init` this with a `folderURL` that is being used elsewhere then `clearAll` will delete not only the cached items, but everything else in the directory. This "gotcha" should either be documented, or ideally you could avoid this by actually caching files in a new directory inside `folderURL`